### ISSUE TITLE
Set random seed of tokenisation

### ISSUE THREAD

**Issue by nsorros** [huggingface/tokenizers#668]
Hi, I noticed that some or all the tokenisation algorithms are not deterministic. Is there a way to set a random seed to have reproducible results? This is more of a problem in smaller datasets I have noticed. Below is some code to reproduce the differences I want to standardise.

```
from tokenizers.models import WordPiece
from tokenizers.normalizers import Lowercase, Sequence
from tokenizers.trainers import WordPieceTrainer
from tokenizers.pre_tokenizers import Whitespace
from tokenizers import Tokenizer
from datasets import load_dataset

vocab_size=30_000
unk_token="[UNK]"

dataset = load_dataset("ag_news")
texts = [example["text"] for example in dataset["train"]]

def tokenize(texts):
    tokenizer = Tokenizer(WordPiece(unk_token=unk_token))
    tokenizer.normalizer = Sequence([Lowercase()])
    tokenizer.pre_tokenizer = Whitespace()

    trainer = WordPieceTrainer(
        vocab_size=vocab_size,
        special_tokens=[unk_token]
    )
    tokenizer.train_from_iterator(texts, trainer=trainer)

    encodings= tokenizer.encode_batch(texts)
    return [e.ids for e in encodings]

texts_encodings_1 = tokenize(texts)
texts_encodings_2 = tokenize(texts)

diffs = 0
for t1, t2 in zip(texts_encodings_1, texts_encodings_2):
    if t1 != t2:
        diffs += 1

print(f"differences: {diffs} out of {len(texts)}")
```

I get

```
Using custom data configuration default
Reusing dataset ag_news (/Users/nsorros/.cache/huggingface/datasets/ag_news/default/0.0.0/fb5c5e74a110037311ef5e904583ce9f8b9fbc1354290f97b4929f01b3f48b1a)
[00:00:06] Pre-processing sequences                 ██████████████████████████████████████████████████████████████████████████████████████████████████████████ 0        /        0
[00:00:00] Tokenize words                           ██████████████████████████████████████████████████████████████████████████████████████████████████████████ 65353    /    65353
[00:00:00] Count pairs                              ██████████████████████████████████████████████████████████████████████████████████████████████████████████ 65353    /    65353
[00:00:01] Compute merges                           ██████████████████████████████████████████████████████████████████████████████████████████████████████████ 29889    /    29889

[00:00:06] Pre-processing sequences                 ██████████████████████████████████████████████████████████████████████████████████████████████████████████ 0        /        0
[00:00:00] Tokenize words                           ██████████████████████████████████████████████████████████████████████████████████████████████████████████ 65353    /    65353
[00:00:00] Count pairs                              ██████████████████████████████████████████████████████████████████████████████████████████████████████████ 65353    /    65353
[00:00:01] Compute merges                           ██████████████████████████████████████████████████████████████████████████████████████████████████████████ 29889    /    29889

differences: 30588 out of 120000
```
